src/_pytest/config/__init__.py

```diff
@@ -267,10 +267,13 @@ def parse_hookimpl_opts(self, plugin, name):
         # collect unmarked hooks as long as they have the `pytest_' prefix
         if opts is None and name.startswith("pytest_"):
             opts = {}
-
         if opts is not None:
+            # TODO: DeprecationWarning, people should use hookimpl
+            known_marks = {m.name for m in getattr(method, "pytestmark", [])}
+
             for name in ("tryfirst", "trylast", "optionalhook", "hookwrapper"):
-                opts.setdefault(name, hasattr(method, name))
+
+                opts.setdefault(name, hasattr(method, name) or name in known_marks)
```

Comment:
Oh completely forgot it was even possible to use marks to define hookimpl options. ^^